src/mtype.d

```diff
@@ -4542,103 +4542,23 @@ extern (C++) class TypeArray : TypeNext
         }
         if (ident == Id.reverse && (n.ty == Tchar || n.ty == Twchar))
         {
-            static __gshared const(char)** reverseName = ["_adReverseChar", "_adReverseWchar"];
-            static __gshared FuncDeclaration* reverseFd = [null, null];
-
-            deprecation(e.loc, "use std.algorithm.reverse instead of .reverse property");
-            int i = n.ty == Twchar;
-            if (!reverseFd[i])
-            {
-                auto params = new Parameters();
-                Type next = n.ty == Twchar ? Type.twchar : Type.tchar;
-                Type arrty = next.arrayOf();
-                params.push(new Parameter(0, arrty, null, null));
-                reverseFd[i] = FuncDeclaration.genCfunc(params, arrty, reverseName[i]);
-            }
-
-            Expression ec = new VarExp(Loc(), reverseFd[i], false);
-            e = e.castTo(sc, n.arrayOf()); // convert to dynamic array
-            auto arguments = new Expressions();
-            arguments.push(e);
-            e = new CallExp(e.loc, ec, arguments);
-            e.type = next.arrayOf();
+            error(e.loc, "use std.algorithm.reverse instead of .reverse property");
```

Comment:
Since this is a unique error message and not the generic "undefined identifier". Shouldn't there be a test for that, make sure it fails correctly?